### PR TITLE
🧃 Add the Proposals filters component

### PR DESCRIPTION
**Pull request by thesan**
See #585

### CODE REVIEW

File: packages/ui/src/proposals/components/ProposalFilters/ProposalFilters.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        />
+
+        <SimpleSelect
+          title="Proposer"
+          options={proposersOptions}
+          value={proposer}
+          onChange={(value) => {
+            dispatch({ type: 'change', field: 'proposer', value })
+          }}
+        />
```
[the remark this answers is not in the document]
Comment (jodator) [reply packages/ui/src/proposals/components/ProposalFilters/ProposalFilters.tsx:122]: > Also would it be fine to reuse the `<MockApolloProvider />` from the testing code, to render `<SelectMember />` in storybook ?

I think that we could use that 👍

File: packages/ui/src/common/utils.ts [hunk 2]
```diff
@@ -3,3 +3,12 @@ export const isFunction = (something: unknown): something is CallableFunction =>
 export const isDefined = <T extends any>(something: T | undefined): something is T => typeof something !== 'undefined'
 
 export const isNumber = (something: unknown): something is number => typeof something === 'number'
+
+export const objectShallowContains = <T extends { [k: string]: any }>(contained: T): ((container: T) => boolean) => {
```
Comment (jodator) [inline packages/ui/src/common/utils.ts:7]: I think that the name might be misleading. AFAICS it compares an object to be shallowly equal to others, so maybe just `objectEquals()`? 

Below some variables names hint, but you can use any other you see fit.

```suggestion
export const objectEquals = <T extends Record<string, any>}>(reference: T): ((compared: T) => boolean) => {
```

**Review by jodator — CHANGES_REQUESTED** [review]
Hi, I think that the clear filters change might enforce other changes in the code so I'll stop here. But, apart other things that I've commented the PR looks good.

File: packages/ui/src/common/utils.ts (hunk 2 above)
Comment (thesan) [reply packages/ui/src/common/utils.ts]: Hi @jodator, thanks for the review.

Actually it was only doing something similar to Jest  [`expect.objectContaining`](https://jestjs.io/docs/expect#expectobjectcontainingobject). But I guess it makes more sense to check the keys length of the `compared` object to have something easier and safer to reuse somewhere else. So I changed the logic a bit.

**Review by jodator — APPROVED** [review]

**Pull request merged**